assets/scripts/segments/SegmentCanvas.jsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import { getSegmentVariantInfo } from './info'
+import { drawSegmentContents, getVariantInfoDimensions } from './view'
+import { TILE_SIZE } from './constants'
+
+const WIDTH_PALETTE_MULTIPLIER = 4 // Dupe from palette.js
+const SEGMENT_Y_NORMAL = 265
+const SEGMENT_Y_PALETTE = 20
+const CANVAS_HEIGHT = 480
+const CANVAS_GROUND = 35
+const CANVAS_BASELINE = CANVAS_HEIGHT - CANVAS_GROUND
+
+class SegmentCanvas extends React.Component {
+  static propTypes = {
+    width: PropTypes.number.isRequired,
+    type: PropTypes.string.isRequired,
+    variantString: PropTypes.string.isRequired,
+    forPalette: PropTypes.bool,
+    randSeed: PropTypes.number,
+    dpi: PropTypes.number
+  }
+
+  componentDidMount () {
+    this.drawSegment()
+  }
+
+  componentDidUpdate (prevProps) {
+    if (!this.props.forPalette) {
+      this.drawSegment()
+    }
+  }
+
+  drawSegment = () => {
+    const multiplier = this.props.forPalette ? (WIDTH_PALETTE_MULTIPLIER / TILE_SIZE) : 1
+    const offsetTop = this.props.forPalette ? SEGMENT_Y_PALETTE : SEGMENT_Y_NORMAL
+    const ctx = this.canvasEl.getContext('2d')
+    ctx.clearRect(0, 0, this.canvasEl.width, this.canvasEl.height)
+    drawSegmentContents(ctx, this.props.type, this.props.variantString, this.props.width, 0, offsetTop, this.props.randSeed, multiplier, this.props.forPalette)
+  }
+
+  render () {
+    const variantInfo = getSegmentVariantInfo(this.props.type, this.props.variantString)
+    const multiplier = this.props.forPalette ? (WIDTH_PALETTE_MULTIPLIER / TILE_SIZE) : 1
+    const dimensions = getVariantInfoDimensions(variantInfo, this.props.width, multiplier)
+    const totalWidth = dimensions.right - dimensions.left
+
+    const canvasWidth = this.props.forPalette ? this.props.width * this.props.dpi : totalWidth * TILE_SIZE * this.props.dpi
+    const canvasHeight = CANVAS_BASELINE * this.props.dpi
+    const canvasStyle = {
+      width: this.props.forPalette ? this.props.width : totalWidth * TILE_SIZE,
+      height: CANVAS_BASELINE,
+      left: (dimensions.left * TILE_SIZE * multiplier)
+    }
+
+    return (
+      <div>
```

Comment:
Do we need this `<div>` wrapper?

Reply:
Without the `<div>` wrapper, the CSS styles don't get applied correctly. I tried changing the class names for the switching-in and switching-away styles in the stylesheet but it doesn't appear to work.

Reply:
I decided to move the wrapper `<div>` to the `renderSegmentCanvas` function in the Segment component. This wrapper will be needed when I connect the canvas to react-dnd and also I can't seem to figure out how to apply the switching away transitions without that wrapper `<div>`. 

Also I moved the hover `<div>` from the SegmentCanvas to the Segment component since we only need one hover background.